Source/JavaScriptCore/b3/B3Common.cpp

```diff
@@ -70,10 +70,10 @@ bool shouldSaveIRBeforePhase()
     return Options::verboseValidationFailure();
 }
 
-std::optional<GPRReg> pinnedExtendedOffsetAddrRegister()
+std::optional<GPRReg> linkRegister()
 {
-#if CPU(ARM64) || CPU(RISCV64)
-    return MacroAssembler::dataTempRegister;
+#if CPU(ARM64)
+    return MacroAssembler::linkRegister;
 #elif CPU(X86_64)
     return std::nullopt;
 #else
```

Comment:
Can we just ASSERT that this is only ever called on arm64 now? Also, if we do that, no need to return optional. Just return a GPRReg now, since on arm64 it'll be correct. And we'll crash on non arm64

Reply:
updated